### PR TITLE
IGNITE-14495 Add common internal exceptions.

### PR DESCRIPTION
**Pull request by ascherbakoff**

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/lang/IgniteInternalCheckedException.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        super(msg, cause, true, writableStackTrace);
+    }
+
+    /**
+     * Creates new exception with given error message and optional nested exception.
+     *
+     * @param msg Error message.
+     * @param cause Optional nested exception (can be {@code null}).
+     */
+    public IgniteInternalCheckedException(String msg, Throwable cause) {
```
Comment (AMashenkov) [inline modules/core/src/main/java/org/apache/ignite/lang/IgniteInternalCheckedException.java:70]:
```suggestion
     * @param cause Optional nested exception.
     */
    public IgniteInternalCheckedException(String msg, @Nullable Throwable cause) {
```

File: modules/core/src/main/java/org/apache/ignite/lang/IgniteInternalCheckedException.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+/**
+ * General internal checked exception. This exception is used to indicate any error condition within the node.
+ */
+public class IgniteInternalCheckedException extends Exception {
+    /** */
+    private static final long serialVersionUID = 0L;
+
+    /**
+     * Create empty exception.
```
Comment (sashapolo) [inline modules/core/src/main/java/org/apache/ignite/lang/IgniteInternalCheckedException.java:27]:
```suggestion
     * Creates an empty exception.
```

File: modules/core/src/main/java/org/apache/ignite/lang/IgniteInternalCheckedException.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+     */
+    public IgniteInternalCheckedException() {
+        // No-op.
+    }
+
+    /**
+     * Creates new exception with given error message.
```
Comment (sashapolo) [inline modules/core/src/main/java/org/apache/ignite/lang/IgniteInternalCheckedException.java:34]:
```suggestion
     * Creates a new exception with the given error message.
```

File: modules/core/src/main/java/org/apache/ignite/lang/IgniteInternalCheckedException.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+     *
+     * @param msg Error message.
+     */
+    public IgniteInternalCheckedException(String msg) {
+        super(msg);
+    }
+
+    /**
+     * Creates new grid exception with given throwable as a cause and
```
Comment (sashapolo) [inline modules/core/src/main/java/org/apache/ignite/lang/IgniteInternalCheckedException.java:43]:
```suggestion
     * Creates a new exception with the given throwable as the cause and the 
```

File: modules/core/src/main/java/org/apache/ignite/lang/IgniteInternalCheckedException.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+     * source of error message.
+     *
+     * @param cause Non-null throwable cause.
+     */
+    public IgniteInternalCheckedException(Throwable cause) {
+        this(cause.getMessage(), cause);
+    }
+
+    /**
+     * Creates new exception with given error message and optional nested exception.
```
Comment (sashapolo) [inline modules/core/src/main/java/org/apache/ignite/lang/IgniteInternalCheckedException.java:53]:
```suggestion
     * Creates a new exception with the given error message and an optional nested exception.
```

File: modules/core/src/main/java/org/apache/ignite/lang/IgniteInternalCheckedException.java [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+     *
+     * @param msg Error message.
+     * @param cause Optional nested exception (can be {@code null}).
+     * @param writableStackTrace whether or not the stack trace should
+     *                           be writable
+     */
+    public IgniteInternalCheckedException(String msg, Throwable cause, boolean writableStackTrace) {
```
Comment (sashapolo) [inline modules/core/src/main/java/org/apache/ignite/lang/IgniteInternalCheckedException.java:60]:
```suggestion
    public IgniteInternalCheckedException(String msg, @Nullable Throwable cause, boolean writableStackTrace) {
```

File: modules/core/src/main/java/org/apache/ignite/lang/IgniteInternalCheckedException.java [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+        super(msg, cause, true, writableStackTrace);
+    }
+
+    /**
+     * Creates new exception with given error message and optional nested exception.
```
Comment (sashapolo) [inline modules/core/src/main/java/org/apache/ignite/lang/IgniteInternalCheckedException.java:65]:
```suggestion
     * Creates a new exception with the given error message and an optional nested exception.
```

File: modules/core/src/main/java/org/apache/ignite/lang/IgniteInternalCheckedException.java [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+     */
+    public IgniteInternalCheckedException(String msg, Throwable cause) {
+        super(msg, cause);
+    }
+
+    /** {@inheritDoc} */
+    @Override public String toString() {
+        return getClass() + ": " + getMessage();
+    }
+}
```
Comment (sashapolo) [inline modules/core/src/main/java/org/apache/ignite/lang/IgniteInternalCheckedException.java:78]: missing line break

**Comment by ibessonov** [comment]
Merged manually

**Pull request closed without merging**